src/account/interface.ts

```diff
@@ -100,6 +119,24 @@ export abstract class AccountInterface extends ProviderInterface {
     transactionsDetail?: InvocationsDetails
   ): Promise<DeclareContractResponse>;
 
+  /**
+   * Deploy the account on Starknet
+   * @param contractPayload transaction payload to be deployed containing:
+   * - classHash: computed class hash of compiled contract
+   * - constructor calldata
+   * - address salt  
+  - signature
+   * @param transactionsDetail Invocation Details containing:
+  - optional nonce
+  - optional version
+  - optional maxFee
+   * @returns a confirmation of sending a transaction on the starknet contract
+   */
+  public abstract deployAccount(
+    contractPayload: DeployAccountContractTransaction,
```

Comment:
Turns out, this was a typo. It has to be `DeployAccountContractPayload`

Reply:
Yea it is a typo, but also as I was exploring it, it came to me that we separate transactions into payload + details (nonce, signature,fee). But by starkware specs it is one transaction, we invented details in starknetjs for no specific reason, and I can't find out why, so this can just confuse. 

I will create a separate Issue to resolve this.